server/api_webex.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/mattermost/mattermost-server/model"
+)
+
+// https://developer.webex.com/docs/api/getting-started
+// https://github.com/webex/react-ciscospark
+
+func (p *Plugin) handleGetWebexUser(w http.ResponseWriter, r *http.Request) {
+	sessionUserID := r.Header.Get("Mattermost-User-ID")
+	if sessionUserID == "" {
+		JSONErrorResponse(w, ErrNotAuthorized, http.StatusUnauthorized)
+		return
+	}
+
+	user, err := p.loadWebexUser(sessionUserID)
+	if err != nil {
+		JSONResponse(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	JSONResponse(w, user, http.StatusOK)
+}
+
+type StartMeetingRequest struct {
+	ChannelId string `json:"channel_id"`
+	Personal  bool   `json:"personal"`
+	Topic     string `json:"topic"`
+	MeetingId int    `json:"meeting_id"`
+}
+
+func (p *Plugin) handleStartMeeting(w http.ResponseWriter, r *http.Request) {
+
+	sessionUserID := r.Header.Get("Mattermost-User-Id")
+	if sessionUserID == "" {
+		JSONErrorResponse(w, ErrNotAuthorized, http.StatusUnauthorized)
+		return
+	}
+
+	var req StartMeetingRequest
+
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		JSONErrorResponse(w, err, http.StatusBadRequest)
+		return
+	}
+
+	_, err := p.loadWebexSession(sessionUserID)
+	switch {
+	case
+		err == ErrWebexSessionNotFound,
+		err == ErrWebexUserNotFound:
+
+		p.API.LogError("Error retrieving session", "err", err.Error())
+		JSONErrorResponse(w, ErrWebexNotConnected, http.StatusUnauthorized)
+		return
+
+	case err != nil:
+		JSONErrorResponse(w, err, http.StatusInternalServerError)
+		return
+
+	}
+
+	user, err := p.loadWebexUser(sessionUserID)
+	switch {
+	case
+		err == ErrWebexSessionNotFound,
+		err == ErrWebexUserNotFound:
+		JSONErrorResponse(w, ErrWebexNotConnected, http.StatusUnauthorized)
+		return
+	case err != nil:
+		JSONErrorResponse(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	channel, appErr := p.API.GetChannel(req.ChannelId)
+	if appErr != nil {
+		JSONErrorResponse(w, appErr, http.StatusInternalServerError) // maybe bad request
+		return
+	}
+
+	fmt.Printf("\n%+v\n\n", channel)
+
+	p.API.LogDebug("channel", "channel_id", channel.Id, "name", channel.Name, "session_user", sessionUserID, "type", channel.Type)
+
+	// Not sure how reliable this is...
```

Comment:
Nop, actually is what we do in our code. I would do something like checking the channel type before, to make sure with: `channel.Type == model.CHANNEL_DIRECT`

Reply:
Indeed, but it's annoying that we don't at least have a helper method in `model` to simplify this...

Question: does webex not support calls among multiple people? Could we simplify this code to just fetch the channel members and support non-direct message channels too?

Reply:
@jespino Thanks! I'll update it to use the Type check.  That'll make it easier when this is extended to support multiple people.

@lieut-data Webex does support multiple people but it's handled differently than 1:1 by using more of a room concept.  The flow is different enough that I opted to leave out the functionality on this initial pass.  Since they [webex] seem to expect an integration to be embedded within applications I didn't want to build more around the current basic solution with the html/js page.  In case this integration would be preferred by the UI/UX folks to be a more embedded part of the MM application(s).